Fatras/include/ActsFatras/EventData/DigitizedHit.hpp

```diff
@@ -0,0 +1,82 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Surfaces/Surface.hpp"
+#include "ActsFatras/EventData/Hit.hpp"
+
+namespace ActsFatras {
+
+/// A digitizeed hit on a surface.
+///
+/// This is the digitized hit, output of any digitization module. It serves
+/// as source_link in the Surface based Measurement classes and thus has to
+/// fullfill the SourceLinkConcept.
+///
+class DigitizedHit {
+ public:
+  /// @brief Nested IContent base class
+  ///
+  /// This will be overwritten by specialisations of the
+  /// varius digitisers that will know their own content
+  class IContent {
+   public:
+    virtual ~IContent() = default;
+  };
+
+  DigitizedHit() = default;
+
+  /// Constructor with arguments
+  ///
+  /// @param hits The simulated hits at input
+  /// @param surface The surface for the measurement representation
+  /// @param content The digitization content to be filled by the digitizer
+  DigitizedHit(std::vector<Hit>&& hits, const Acts::Surface& surface,
+               std::unique_ptr<const DigitizedHit::IContent> content);
+
+  virtual ~DigitizedHit() = default;
+
+  /// Required copy constructor
+  ///
+  /// @param other The source
+  DigitizedHit(const DigitizedHit& other);
+
+  /// Required equality operator
+  ///
+  /// @param other The one to compare to
+  bool operator==(const DigitizedHit& other) const;
```

Comment:
This is required by the `SourceLinkConcept`.

Reply:
I see. I understand that is it required, but I still think it is ill-defined.

This may be a bit of a tangent, but I have the same feeling for a lot of the value comparisons that are implemented elsewhere in the code. E.g. track parameters can be value-compared, but depending on the context I might want to have identity comparison, comparison within a given uncertainty, comparison with the given covariances, or comparison of the effective global parameters (in which case you could have two parameters defined on different surfaces being equal).

Reply:
Not relevant anymore for this PR. `DigitizedHit` is gone.